src/api/app/models/worker_status.rb

```diff
@@ -37,29 +37,73 @@ def project_names(prjs)
     end
   end
 
-  def update_workerstatus_cache
-    # do not add hiding in here - this is purely for statistics
-    backend_status = Backend::Api::BuildResults::Worker.status
-    wdata = Nokogiri::XML(backend_status)
+  def save
+    @workerstatus = Rails.cache.read('workerstatus')
+    return unless @workerstatus
+
     @mytime = Time.now.to_i
     @squeues = Hash.new(0)
 
-    Rails.cache.write('workerstatus', backend_status, expires_in: 3.minutes)
-
     StatusHistory.transaction do
       ['blocked', 'waiting'].each do |state|
-        wdata.search("//#{state}").each { |e| save_value_line(e, state) }
+        @workerstatus.search("//#{state}").each { |e| save_value_line(e, state) }
       end
-      wdata.search('partition/daemon').each { |daemon| parse_daemon_infos(daemon) }
-      parse_worker_infos(wdata)
+      @workerstatus.search('partition/daemon').each { |daemon| parse_daemon_infos(daemon) }
+      parse_worker_infos(@workerstatus)
       @squeues.each_pair { |key, value| StatusHistory.create(time: @mytime, key: key, value: value) }
     end
 
-    backend_status
+    send_to_bus
   end
 
   private
 
+  def send_to_bus
```

Comment:
could break this method in a service?

Reply:
After sleeping on it it might be more clever to do this in a job with another frequency. Every 30 seconds might be a bit too much.